src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/RazorParserOptions.cs

```diff
@@ -191,4 +192,15 @@ public RazorParserOptions WithFlags(
             ? this
             : new(LanguageVersion, FileKind, Directives, CSharpParseOptions, flags);
     }
+
+    public bool Equals(RazorParserOptions? other)
+    {
+        return
+            other is not null
+            && LanguageVersion == other.LanguageVersion
+            && FileKind == other.FileKind
+            && CSharpParseOptions == other.CSharpParseOptions
+            && _flags == other._flags
```

Comment:
Consider reordering `_flags` and `FileKind` earlier to push the cheaper equality checks to the front.